src/libraries/System.Private.CoreLib/src/System/Threading/NativeRuntimeEventSource.PortableThreadPool.cs

```diff
@@ -183,6 +171,10 @@ private unsafe void WriteThreadEvent(int eventId, uint numExistingThreads)
             data[3].Size = sizeof(ushort);
             data[3].Reserved = 0;
             WriteEventCore(55, 4, data);
+
+#if FEATURE_EVENTSOURCE_XPLAT
+            XplatEventLogger.ThreadPoolWorkerThreadAdjustmentAdjustment(AverageThroughput, NewWorkerThreadCount, Reason, ClrInstanceID);
+#endif
```

Comment:
I could use some education... why is this needed, rather than for example registering a listener that routes events to LTTng?

Also, moving forward is LTTng a path we're still excited about / supporting?  I assumed with event pipe and the rest of the dotnet tooling around tracing cross-platform that we'd be deprecating this support.

Reply:
We do still want to support LTTng, as we're seeing some amount of uptake across other Microsoft products, and we want to be able to participate in this.  That said, this code is essentially making it look like the thread pool is still implemented in native code.  So rather than using an EventSource in managed code, this is invoking the native FireETW* functions, which result in calls to EventPipe, ETW, and/or LTTng.  So, this code is actually platform agnostic, and takes advantage of the platform-specific logger code in the runtime itself.

That said, thinking about this a bit further, I am wondering if there's something that we can't do here in a more generic direction.  Rather than having to create a new method and QCALL for every event, can we just use the `NativeRuntimeEventSource` as is, and modify `EventPipe` to be OK with multiple provider registrations (at least for the runtime provider)?  I could believe that over time there will be more components that move to managed code, and we'll want to keep logging the same events.  ETW will already allow multiple provider registrations.  LTTng might need something special - possibly some pre-generated code that would translate between the `EventSource` and the LTTng tracepoints.  But, I'd be tempted to say that we shouldn't do something special for it, and just treat the events like any other `EventSource` events with respect to LTTng.

Reply:
You're right - this is calling `FireEtXPlat` which is beyond the decision of which logger to call - this will call LTTng on Linux directly.  You also bring up a good point that we may be double emitting.

Yes, I was proposing that we not worry about the difference in how the events are emitted when they are moved to managed code.  If we wanted to be more aggressive about maintaining perfect compatibility, we could also choose to build  auto-generated code that would allow us to call any of the events and have them emitted as a strongly typed LTTng event.  In the end, I think that the thing I want to encourage is that it would be nice if we didn't do special casing for these events here.  I suspect that we'll end up doing more of this type of thing, and it would be great if calling into `NativeRuntimeEventSource` to emit an event "just worked".

Reply:
Thanks for the suggestions @noahfalk and @brianrob. I went and checked and indeed I this causes it to double-emit events for EventListener. For LTTng it ends up being emitted as an "EventSourceEvent" event as well as the individual threadpool event. I'll fix this to use FireETW* events instead like @noahfalk suggested. 

> Yes, I was proposing that we not worry about the difference in how the events are emitted when they are moved to managed code.

I think this wouldn't really be a great experience for devs trying to parse an LTTng trace for a specific type of event by filtering via the name, so that was the main reason I added all these glue to make it seem indifferent from when we had the native threadpool. 

But I also agree with your sentiment towards having some kind of auto-generated code that'll allow devs to add events to ClrEtwAll.man and call it from managed side, and the whole thing "just works". I can try to tackle that later by modifying the scripts to autogenerate the QCall helpers and forward to corresponding LTTng sink.